pallets/loans/src/functions.rs

```diff
@@ -227,6 +220,8 @@ impl<T: Config> Pallet<T> {
 					active_loan.normalized_debt,
 				)?;
 
+				T::InterestAccrual::unreference_rate(active_loan.interest_rate_per_sec)?;
```

Comment:
Maybe do an 

`if active_loan.interest_rate_per_sec != interest_rate_per_sec` and then only update interest rate. Minor but safes one storage reads and writes.

Reply:
That would require splitting up some of the logic of computing whether the rate is valid and then referencing it later. I kind of like that the API exposed by interest-accrual is currently atomic in that regard (except for penalty rates, of course)